.github/workflows/deploy-prod.yml

```
@@ -95,6 +94,20 @@ jobs:
             echo "- Prefer a digest form like \`ghcr.io/org/repo-api@sha256:...\` for production." >> "$GITHUB_STEP_SUMMARY"
           fi
 
+      - name: Refresh GHCR credentials (optional)
+        if: startsWith(env.IMAGE_REF, 'ghcr.io/') && secrets.GHCR_TOKEN != ''
```

actionlint (1.7.11)

[error] context "secrets" is not allowed here. available contexts are "env", "github", "inputs", "job", "matrix", "needs", "runner", "steps", "strategy", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)